### PR TITLE
SPE Compat - Add Tank cannon overpressure and shell fragmentation

### PR DESCRIPTION
**Pull request by jpspider1**
**When merged this pull request will:**
- HE and AP shells (for AP shells only those that would have had filler and aren't solid shot) for all German and American Tanks will have Fragmentation values added
- New shells.hpp created to encompass all shell fragmentations
- 2cm Flak bullet has fragmentation under bullet.hpp
- CfgAmmo.hpp will include new shells.hpp with changes
- SPE_TankCannon_base will have basic ACE Overpressure values from wiki
- SPE_TankCannon_base added in \compat_spe\CfgWeapons\cannons.hpp

### CODE REVIEW

File: addons/compat_spe/CfgAmmo/bullets.hpp [hunk 1]
```diff
@@ -10,3 +10,28 @@ class SPE_B_127x99_Ball: SPE_Bullet_Vehicle_base {
...
+class SPE_Bullet_AA_base;
+class SPE_SprGr_FlaK_38: SPE_Bullet_AA_base
+{
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 50;
+    EGVAR(frag,charge) = 6;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_tiny","ACE_frag_small_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/bullets.hpp:24]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_tiny", "ACE_frag_small_HD"};
```

File: addons/compat_spe/CfgAmmo/bullets.hpp [hunk 2]
```diff
@@ -10,3 +10,28 @@ class SPE_B_127x99_Ball: SPE_Bullet_Vehicle_base {
...
+class SPE_Bullet_AP_base;
+class SPE_PzGr_FlaK_38_AP_T: SPE_Bullet_AP_base
+{
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 50;
+    EGVAR(frag,charge) = 6;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_tiny","ACE_frag_small_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/bullets.hpp:36]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_tiny", "ACE_frag_small_HD"};
```

File: addons/compat_spe/CfgWeapons/cannons.hpp [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// CANNON
+class SPE_CannonCore;
+class SPE_TankCannon_base: SPE_CannonCore
+{
+    ace_overpressure_angle = 90;    // Cone in which the damage is applied (in degrees from the muzzle of the cannon)
+    ace_overpressure_range = 50;    // Range in meters in which the damage is applied
+    ace_overpressure_damage = 0.85; // Damage multiplier
```
Comment (DartRuffian) [inline addons/compat_spe/CfgWeapons/cannons.hpp:7]:
```suggestion
    EGVAR(overpressure,angle) = 90;    // Cone in which the damage is applied (in degrees from the muzzle of the cannon)
    EGVAR(overpressure,range) = 50;    // Range in meters in which the damage is applied
    EGVAR(overpressure,damage) = 0.85; // Damage multiplier
```

File: addons/compat_spe/CfgWeapons/cannons.hpp [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// CANNON
+class SPE_CannonCore;
+class SPE_TankCannon_base: SPE_CannonCore
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgWeapons/cannons.hpp:4]:
```suggestion
class SPE_TankCannon_base: SPE_CannonCore {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+//M1 57mm AT gun
+class SPE_6pdr_mk10T_HE: SPE_ShellHE_base
+{
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 2720;
+    EGVAR(frag,charge) = 590;
+    EGVAR(frag,gurney_c) = 2440;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_small_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:129]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_medium", "ACE_frag_small_HD"};
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,metal) = 5840;
+    EGVAR(frag,charge) = 390;
+    EGVAR(frag,gurney_c) = 2440;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
+};
+
+//M1 57mm AT gun
+class SPE_6pdr_mk10T_HE: SPE_ShellHE_base
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:122]:
```suggestion
// M1 57mm AT gun
class SPE_6pdr_mk10T_HE: SPE_ShellHE_base {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,metal) = 5840;
+    EGVAR(frag,charge) = 390;
+    EGVAR(frag,gurney_c) = 2440;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
+};
+
+//M4 Sherman 76 HE
+class SPE_76mm_M1_M42_HE: SPE_76mm_M7_M42_HE
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:110]:
```suggestion
// M4 Sherman 76 HE
class SPE_76mm_M1_M42_HE: SPE_76mm_M7_M42_HE {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,metal) = 5840;
+    EGVAR(frag,charge) = 390;
+    EGVAR(frag,gurney_c) = 2440;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
+};
+
+//M18 HE
+class SPE_76mm_M1_HC_M42_HE: SPE_76mm_M7_M42_HE
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:98]:
```suggestion
// M18 HE
class SPE_76mm_M1_HC_M42_HE: SPE_76mm_M7_M42_HE {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 5840;
+    EGVAR(frag,charge) = 390;
+    EGVAR(frag,gurney_c) = 2440;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:105]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_medium", "ACE_frag_medium_HD"};
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+//M10 HE
+class SPE_76mm_M7_M42_HE: SPE_ShellHE_base
+{
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 5840;
+    EGVAR(frag,charge) = 390;
+    EGVAR(frag,gurney_c) = 2440;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:93]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_medium", "ACE_frag_medium_HD"};
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,metal) = 9400;
+    EGVAR(frag,charge) = 1000;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_large","ACE_frag_large_HD"};
+};
+
+//M10 HE
+class SPE_76mm_M7_M42_HE: SPE_ShellHE_base
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:86]:
```suggestion
// M10 HE
class SPE_76mm_M7_M42_HE: SPE_ShellHE_base {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,metal) = 14810;
+    EGVAR(frag,charge) = 2530;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_large","ACE_frag_large_HD"};
+};
+
+//Nashorn HE
+class SPE_SprGr39_HE: SPE_ShellHE_base
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:74]:
```suggestion
// Nashorn HE
class SPE_SprGr39_HE: SPE_ShellHE_base {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 9400;
+    EGVAR(frag,charge) = 1000;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_large","ACE_frag_large_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:81]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_large", "ACE_frag_large_HD"};
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+//StuH 42 HE
+class SPE_105mm_StuH42_Gr38_HE: SPE_ShellHE_base
+{
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 14810;
+    EGVAR(frag,charge) = 2530;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_large","ACE_frag_large_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:69]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_large", "ACE_frag_large_HD"};
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,metal) = 9000;
+    EGVAR(frag,charge) = 900;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_large","ACE_frag_large_HD"};
+};
+
+//StuH 42 HE
+class SPE_105mm_StuH42_Gr38_HE: SPE_ShellHE_base
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:62]:
```suggestion
// StuH 42 HE
class SPE_105mm_StuH42_Gr38_HE: SPE_ShellHE_base {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+//Panzer VI HE
+class SPE_SprGr_KwK36_HE: SPE_ShellHE_base
+{
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 9000;
+    EGVAR(frag,charge) = 900;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_large","ACE_frag_large_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:57]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_large", "ACE_frag_large_HD"};
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,metal) = 5740;
+    EGVAR(frag,charge) = 725;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
+};
+
+//Panzer VI HE
+class SPE_SprGr_KwK36_HE: SPE_ShellHE_base
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:50]:
```suggestion
// Panzer VI HE
class SPE_SprGr_KwK36_HE: SPE_ShellHE_base {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,metal) = 5740;
+    EGVAR(frag,charge) = 686;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
+};
+
+//Panzer V HE
+class SPE_SprGr42_KwK42_HE: SPE_ShellHE_base
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:38]:
```suggestion
// Panzer V HE
class SPE_SprGr42_KwK42_HE: SPE_ShellHE_base {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 5740;
+    EGVAR(frag,charge) = 725;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:45]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_medium", "ACE_frag_medium_HD"};
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 20]
```diff
@@ -0,0 +1,130 @@
...
+//Panzer IV, StuG III G HE
+class SPE_SprGr34_KWK40_HE: SPE_ShellHE_base
+{
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 5740;
+    EGVAR(frag,charge) = 686;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:33]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_medium", "ACE_frag_medium_HD"};
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 21]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 1820;
+    EGVAR(frag,charge) = 217;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_small_HD"};
+};
+//Panzer III N HE
+class SPE_SprGr34_K51_HE: SPE_ShellHE_base
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:14]:
```suggestion
// Panzer III N HE
class SPE_SprGr34_K51_HE: SPE_ShellHE_base {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 22]
```diff
@@ -0,0 +1,130 @@
+//Panzer III J, L, M HE
+class SPE_50mm_SprGr38_HE: SPE_ShellHE_base
+{
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:3]:
```suggestion
// Panzer III J, L, M HE
class SPE_50mm_SprGr38_HE: SPE_ShellHE_base {
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 23]
```diff
@@ -0,0 +1,130 @@
...
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 1820;
+    EGVAR(frag,charge) = 217;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_small_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:10]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_medium", "ACE_frag_small_HD"};
```

File: addons/compat_spe/CfgAmmo/shells.hpp [hunk 24]
```diff
@@ -0,0 +1,130 @@
...
+//Panzer III N HE
+class SPE_SprGr34_K51_HE: SPE_ShellHE_base
+{
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 5740;
+    EGVAR(frag,charge) = 686;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_medium","ACE_frag_medium_HD"};
```
Comment (DartRuffian) [inline addons/compat_spe/CfgAmmo/shells.hpp:21]:
```suggestion
    EGVAR(frag,classes)[] = {"ACE_frag_medium", "ACE_frag_medium_HD"};
```

**Comment by jpspider1** [comment]
I think I will also add ACE fragmentation parameters to SPE HEAT shells and some APHE shells. I think APCR and solid AP shot will not have any ACE fragmentation.

File: addons/compat_spe/CfgAmmo/bullets.hpp [hunk 25]
```diff
@@ -10,3 +10,28 @@ class SPE_B_127x99_Ball: SPE_Bullet_Vehicle_base {
 class SPE_B_127x99_API: SPE_B_127x99_Ball {
     EGVAR(vehicle_damage,incendiary) = 1;
 };
+
+// FLAK
+class SPE_Bullet_AA_base;
+class SPE_SprGr_FlaK_38: SPE_Bullet_AA_base
+{
```
Comment (PabstMirror) [inline addons/compat_spe/CfgAmmo/bullets.hpp:17]:
```suggestion
class SPE_SprGr_FlaK_38: SPE_Bullet_AA_base {
```
really minor style change

File: addons/compat_spe/CfgAmmo/bullets.hpp [hunk 26]
```diff
@@ -10,3 +10,28 @@ class SPE_B_127x99_Ball: SPE_Bullet_Vehicle_base {
...
+    EGVAR(frag,metal) = 50;
+    EGVAR(frag,charge) = 6;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_tiny", "ACE_frag_small_HD"};
+};
+
+class SPE_Bullet_AP_base;
+class SPE_PzGr_FlaK_38_AP_T: SPE_Bullet_AP_base
+{
```
Comment (PabstMirror) [inline addons/compat_spe/CfgAmmo/bullets.hpp:29]:
```suggestion
class SPE_PzGr_FlaK_38_AP_T: SPE_Bullet_AP_base {
```

**Review by PabstMirror — APPROVED** [review]
everything looks reasonable to me

File: addons/compat_spe/CfgAmmo/bullets.hpp [hunk 27]
```diff
@@ -10,3 +10,26 @@ class SPE_B_127x99_Ball: SPE_Bullet_Vehicle_base {
...
+class SPE_Bullet_AP_base;
+class SPE_PzGr_FlaK_38_AP_T: SPE_Bullet_AP_base {
+    EGVAR(frag,enabled) = 1;
+    EGVAR(frag,force) = 1;
+    EGVAR(frag,metal) = 50;
+    EGVAR(frag,charge) = 6;
+    EGVAR(frag,gurney_c) = 2930;
+    EGVAR(frag,gurney_k) = "1/2";
+    EGVAR(frag,classes)[] = {"ACE_frag_tiny", "ACE_frag_small_HD"};
+};
```
Comment (johnb432) [inline addons/compat_spe/CfgAmmo/bullets.hpp:35]:
```suggestion
};

```

**Pull request merged**